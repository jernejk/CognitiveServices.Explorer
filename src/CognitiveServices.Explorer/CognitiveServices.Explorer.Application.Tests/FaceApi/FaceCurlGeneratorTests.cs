using CognitiveServices.Explorer.Application.FaceApi;
using FluentAssertions;
using Snapshooter.Xunit;
using Xunit;

namespace CognitiveServices.Explorer.Application.Tests.FaceApi
{
    public class FaceCurlGeneratorTests
    {
        [Fact]
        public void ShouldGenerateDetectBinary()
        {
            new FaceCurlGenerator()
                .Detect(new byte[1] { 5 })
                .Should()
                .MatchSnapshot();
        }

        [Fact]
        public void ShouldGenerateDetectUrl()
        {
            new FaceCurlGenerator()
                .Detect("http://test.url")
                .Should()
                .MatchSnapshot();
        }

        [Fact]
        public void ShouldGenerateIdentify()
        {
            string[] faces = { "12234324", "34534253425" };
            new FaceCurlGenerator()
                .Identify("default", faces)
                .Should()
                .MatchSnapshot();
        }
    }
}
