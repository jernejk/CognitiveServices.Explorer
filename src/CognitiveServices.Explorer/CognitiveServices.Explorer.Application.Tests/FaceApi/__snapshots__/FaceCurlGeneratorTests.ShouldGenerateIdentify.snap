{
  "Headers": {},
  "RelativePath": "face/v1.0/identify",
  "HttpMethod": "POST",
  "Body": "{\"personGroupId\":\"default\",\"faceIds\":[\"12234324\",\"34534253425\"],\"maxNumOfCandidatesReturned\":1,\"confidenceThreshold\":0.5}",
  "BinaryContent": null,
  "ContentType": "application/json",
  "ExpectedHtppCode": 200,
  "TokenHeaderName": "Ocp-Apim-Subscription-Key",
  "CognitiveServiceDoc": "https://westus.dev.cognitive.microsoft.com/docs/services/563879b61984550e40cbbe8d/operations/563879b61984550f30395239"
}
